charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -1,6 +1,9 @@
 {{- if .Values.apiGateway.enabled }}
 {{- if not .Values.client.grpc }}{{ fail "client.grpc must be true for api gateway" }}{{ end }}
-{{- if not .Values.apiGateway.image}}{{ fail "apiGateway.image must be set to enable api gateway" }}{{ end }}
+{{- if not .Values.apiGateway.image}}
```

Comment:
I didn't know a better way to offer a warning to users?

Reply:
Do we need to check the consul version here and fail if the version is too low for consul native api gateway?

Reply:
Oh this might be a good idea, I do think we would have to wait to get everything merged in though first? Otherwise we wouldn't be able to use the chart until the new version went out.
Actually now I'm wondering how we would handle that anyway? It's kind of chicken and egg, right?

Reply:
When it comes to checking versions, we've tried to validate the versions of images before and found it to be practically impossible. You can't truly rely on the image tag to verify the version. We have historically avoided it in this Helm chart because of the abundance of edge cases we'd have to cover.

Reply:
So if something is deployed and the image is incorrect, we usually leave that error to appear at runtime when there is a verifiable way to tell what the image _is_.